### ISSUE TITLE
Add support for command-line commands

### ISSUE THREAD

**Issue by DanySK** [DanySK/alchemist#1]
Alchemist should be able to run simulations without loading any UI, just by passing proper options. @jakebeal, could you help me in identifying a feature set?

# Current feature set

## High priority
* ``--nogui`` - does not open an interface, even if a suitable environment is found
* ``--xml FILE`` - loads the required XML file and runs it
* ``--final-time Time``
* ``--final-step Step``

## Mid priority
* ``--attach-monitors [Monitor syntax to be conceived]`` - allows for configuring and attaching loggers to sample data from the current simulation

## Low priority
* ``--exit-on-close`` terminates the simulator once the current simulation is complete
* Add support for batching through the command line. Suggestions on possible required options, and implementation strategie, are welcome.
* ``--home [DIRECTORY]`` chooses where to save persistent files. Defaults to ~/.alchemist

**Comment by jakebeal** [comment]
This is a good starting set, I think.  Somewhere in there (possibly the simulation XML file) there should be the ability to determine what the simulation termination condition is.  Does this already exist?  If so, I'm not currently aware of it...

**Comment by DanySK** [comment]
I'd dare say that if the purpose of the command line is to support immediate simulations (proto-like) or batches, then the termination condition could be specified directly in the command.

Currently there is support in the engine for terminating it:

* if current time is infinite (the system either is in a steady state or the specific time implementation precision is not fine grained enough). This is a forced termination.
* if a specified point in simulated time has been reached
* if a specified number of simulation steps have been computed
* any of the previous two
* by user input or by any monitor input

We could use the last one to implement arbitrary "stoppers".
However, I'd not put such information in the XML: technically that should be a description of an environment, not a description of a simulation (the only thing that is breaking this abstraction is the "random" entry, that is a consequence of a poor design choice).

Would a --final-time [Time] and --final-step [Step] in the command line work in your opinion?

**Comment by DanySK** [comment]
By the way: arbitrary stoppers are already superseeded by ``--attach-monitors [Monitor syntax to be conceived]``

**Comment by DanySK** [comment]
Alchemist 2.0.0 has its own command line and YAML support. This bug is now outdated.
